### PR TITLE
User/rahat/sft 201/create unit tests for send aux trip and send aux status

### PR DESCRIPTION
**Pull request by islamrahat735**

### CODE REVIEW

**Review by JessieG123 — COMMENTED** [review]
Just wondering, did you config your email/username in git? Because your icon doesn't show up.

File: EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/SendAuxStatusTaskTest.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    osMutexRelease_ExpectAndReturn(auxStatusContactorStatusUpdateMutex, 0);
+    osMutexRelease_ExpectAndReturn(auxStatusReadAuxVoltageMutex, 0);
+    osMutexRelease_ExpectAndReturn(auxStatusOrionInterfaceMutex, 0);
+    osDelayUntil_ExpectAndReturn(prevWakeTime + SEND_AUX_STATUS_TASK_FREQ, osOK);
+    sendAuxStatus(&canQueueData, &prevWakeTime);
+
+    TEST_ASSERT_EQUAL_UINT8_ARRAY_MESSAGE(canQueueData.data, expectedCanQueueData.data, 8, "Data bits aren't equal");
+
+
+
```
Comment (JessieG123) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/SendAuxStatusTaskTest.c:64]: Please remove some extra whitespace and just keep like 1 between each.

File: EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/main.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+void tearDown(void)
+{
+}
+
+int main (void)
+{
+
+    runSendAuxStatusTaskTest();
+    
+}
```
Comment (JessieG123) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/main.c:23]: You should have a new line at the end of the file.

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+int runSendAuxTripTaskTest()
+{
+    UNITY_BEGIN();
+
+    RUN_TEST(checkCanQueueDataForAuxTrip);
+    
+
+    return UNITY_END();
+}
```
Comment (JessieG123) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c:61]: Same here, should have a new line at the end of the file

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.h [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+int runSendAuxTripTaskTest();
```
Comment (JessieG123) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.h:3]: Make sure all your files have a new line at the end

**Review by JessieG123 — CHANGES_REQUESTED** [review]

File: EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/SendAuxStatusTaskTest.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+int runSendAuxStatusTaskTest()
+{
+    UNITY_BEGIN();
+
+    RUN_TEST(checkCanQueueDataForAuxStatus);
+
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/SendAuxStatusTaskTest.c:72]: Remove whitespace

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.h (hunk 4 above)
Comment (badeniran) [reply EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.h:3]: Only .c or .h files tho. You don't need new lines for your Makefiles

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/main.c [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+{
+}
+
+void tearDown(void)
+{
+}
+
+int main (void)
+{
+
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/main.c:20]: Whitespace

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/main.c [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    runSendAuxTripTaskTest();
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/main.c:21]: Need to return what this returns as `main` returns an `int` so should be: `return runSendAuxTripTaskTest()`

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+
+
+int runSendAuxTripTaskTest()
+{
+    UNITY_BEGIN();
+
+    RUN_TEST(checkCanQueueDataForAuxTrip);
+
+
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c:59]: Whitespace

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    osMutexAcquire_ExpectAndReturn(auxTripMutex, 100, osOK);
+    osMessageQueuePut_ExpectAndReturn(canTxGatekeeperQueue, &canQueueData, 0, 0, 0);
+    osMutexRelease_ExpectAndReturn(auxTripMutex, 0);
+    osDelayUntil_ExpectAndReturn(prevWakeTime + SEND_AUX_TRIP_TASK_FREQ, osOK);
+
+    sendAuxTrip( &canQueueData, &prevWakeTime);
+
+    TEST_ASSERT_EQUAL_MESSAGE(canQueueData.data[0], actualCanQueueData.data[0], "Data bits at index 0 aren't equal");
+
+
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c:48]: White space

File: EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/main.c [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+void tearDown(void)
+{
+}
+
+int main (void)
+{
+
+    runSendAuxStatusTaskTest();
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/main.c:21]: return the value of this function since main returns an `int`

**Review by badeniran — CHANGES_REQUESTED** [review]

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.h (hunk 4 above)
Comment (JessieG123) [reply EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.h]: Ohhh I see

File: EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/SendAuxStatusTaskTest.c [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    return UNITY_END();
+}
```
Comment (badeniran) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxStatusTaskTest/SendAuxStatusTaskTest.c:69]: New line

**Review by badeniran — APPROVED** [review]
I think it looks good! Great job! As for the new lines at the end of the file @JessieG123 I think `format.sh` might take them out? I'd say just go through and add a new line at the end of the file and if `format.sh` takes them out then don't worry about it. Reasoning for new line here (https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file#:~:text=There%20are%2Fwere%20some%20C,1.2%2C%202.)&text=Perhaps%20for%20historic%20reasons%2C%20because,the%20first%20standard%20was%20written.)

File: EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+osMutexId_t auxTripMutex;
+const CAN_TxHeaderTypeDef baseCanTxHdr;
+osMessageQueueId_t canTxGatekeeperQueue;
+
+AuxTrip auxTrip =
+{
+    .chargeTripDueToHighCellVoltage = 1, .chargeTripDueToHighTemperatureAndCurrent = 0,
+    .chargeTripDueToPackCurrent = 0, .dischargeTripDueToLowCellVoltage = 1,
+    .dischargeTripDueToHighTemperatureAndCurrent = 0, .dischargeTripDueToPackCurrent = 1,
+    .protectionTrip = 0
```
Comment (JessieG123) [inline EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c:20]: @badeniran was this suppose to be like 
```
blah blah = 
{ 
    .blah = 0
      .blah = 0
        .blah = 0
};
```
after running `format.sh`?

Comment (badeniran) [reply EpsilonAuxBmsV2/Tests/tests/SendAuxTripTaskTest/SendAuxTripTaskTest.c:20]: I think if he put each assignment on a new line it would've done that

**Pull request merged**